src/flask_app.py

```diff
@@ -6,18 +6,22 @@
 
 from charm_state import KNOWN_CHARM_CONFIG, CharmState
 from constants import FLASK_ENV_CONFIG_PREFIX
+from secret_storage import SecretStorage
 
 
 class FlaskApp:  # pylint: disable=too-few-public-methods
     """A class representing the Flask application."""
 
-    def __init__(self, charm_state: CharmState):
+    def __init__(self, charm_state: CharmState, secret_storage: SecretStorage):
         """Initialize a new instance of the FlaskApp class.
 
         Args:
             charm_state: The state of the charm that the FlaskApp instance belongs to.
+            secret_storage: The secret storage object associated with the Flask charm, the secret
+                storage must be initialized before calling any methods of the FlaskApp class.
         """
         self._charm_state = charm_state
+        self._secret_storage = secret_storage
```

Comment:
It indirectly makes the app depandant of the charm which I'm not fan
Why not having the secret_key as part of the state ?